libyul/optimiser/Suite.cpp

```diff
@@ -80,6 +81,7 @@ void OptimiserSuite::run(
 	BlockFlattener{}(ast);
 	ControlFlowSimplifier{}(ast);
 	StructuralSimplifier{*_dialect}(ast);
+	ForLoopCondRewriter{}(ast);
```

Comment:
I think this should be done after the control flow simplifier.

In fact, it should probably be done only before the steps that benefit from this representation and then it should be undone again.

Reply:
Ok so it will looks like
```
ForLoopCondRewriter{}(ast);
ExpressionSplitter{}(ast);
// Actual optimization steps that depend on expression splitter
ExpressionJoiner{}(ast);
ForLoopCondRewriterUndo{}(ast);
```
?

Reply:
yes, something like that.